docs/lazy-adr/adr-006-row-propagation.md

```diff
@@ -0,0 +1,56 @@
+# ADR 006: Block Propagation with Rows
+
+## Changelog
+* 24.06.2021
+
+## Context
+It's a long story of relations between Celestia, Tendermint, and consensus block gossiping. Celestia's team discussed multiple ideas, several ADRs were made, and nothing yet was finalized. The ADR is another attempt to bring changes into block gossiping and hopefully successful.
+
+Currently, we inherit the following from Tendermint. Our codebase relies on the blocks Parts notion, which is a piece of an entire serialized block. Those Parts are gossiped between nodes in consensus and committed with PartSetHeader containing a Merkle Root of the Parts. However, Parts gossiping wasn't designed for Celestia blocks.
+
+Celestia comes with a different block representation from Tendermint. It lays out block as a table of data shares, where Rows or Columns can be and should be gossiped instead of Parts, keeping only one system-wide commitment to data.
+
+## Alternative Approaches
+### ["nah it works just don't touch it"](https://ahseeit.com//king-include/uploads/2020/11/121269295_375504380484919_2997236194077828589_n-6586327691.jpg) approach
+
+It turns out that we could fully treat the Tendermint consensus as a black box. However, that means that we need to verify block integrity twice for both PartSetHeader and DAHeader.
+
+#### Pros
+* Less work
+
+#### Cons
+* This isn't adequate in terms of protocol and software design
+* Potentially introduces security issues(implementer can forget/miss secondary data verification)
```

Comment:
Only a problem if we don't remove the PartSetHeader from the BlockID which is an independent concern of how the block is gossiped during consensus.